### ISSUE TITLE
group_by arg_max regression

### ISSUE THREAD

**Issue by ankopainting** [pola-rs/polars#2516]
#### Are you using Python or Rust?

**Python**.

#### What version of polars are you using?

from poetry.lock
```
polars = [
    {file = "polars-0.12.19-cp36-abi3-macosx_10_7_x86_64.whl", hash = "sha256:5191747259f6eaa2484e0a9e30eb5c798be241c2a59e69de77b5b7531ade6987"},
    {file = "polars-0.12.19-cp36-abi3-macosx_10_9_x86_64.macosx_11_0_arm64.macosx_10_9_universal2.whl", hash = "sha256:9060e1ae8148b7542a7377d1c16750576667037f5171ea156294ceeb95fb8bc4"},
    {file = "polars-0.12.19-cp36-abi3-manylinux_2_12_x86_64.manylinux2010_x86_64.whl", hash = "sha256:68951c67110b0e217ec62cfa8c1c8e1af47d24ffe1a66503a809b74884311fb6"},
    {file = "polars-0.12.19-cp36-abi3-win_amd64.whl", hash = "sha256:6f727c0dd9e3db7ef4de99aa5c2f68e4fa4291eaab9653445526cfabd594074a"},
    {file = "polars-0.12.19.tar.gz", hash = "sha256:23f941cb0663bd74ba4f771e7da159d215e8bda0ad72977de51fdb6159cce6bb"},
]
```

#### What operating system are you using polars on?

macos monterey 12.2 (arm)

#### Describe your bug.

high level i'm grouping by users and sorting by most popular book, this used to work and now it doesn't
as originally asked in stack overflow;
https://stackoverflow.com/questions/70431642/in-polars-how-could-i-use-rank-to-get-most-popular-category-per-user

#### What are the steps to reproduce the behavior?

```
#!/usr/bin/env python3

import polars as pl
import pprint

df = pl.DataFrame({'book': ['bookA',
          'bookA',
          'bookB',
          'bookA',
          'bookA',
          'bookC',
          'bookC',
          'bookC'],
 'transaction_id': [1, 2, 3, 4, 5, 6, 7, 8],
 'user': ['bob', 'bob', 'bob', 'tim', 'lucy', 'lucy', 'lucy', 'lucy']
})

df = (df.groupby(["user", "book"])
 .agg([
     pl.col("book").count()
 ])
 .groupby("user")
 .agg([
     pl.col("book").take(pl.col("book_count").arg_max()).alias("fav_book")
 ])
)

print(df)
```

#### What is the actual behavior?

there are two curious things about the output

```
(base) ➜  polars_poetry poetry run python3 book_agg.py
shape: (3, 2)
┌──────┬─────────────────────────────┐
│ user ┆ fav_book                    │
│ ---  ┆ ---                         │
│ str  ┆ list [str]                  │
╞══════╪═════════════════════════════╡
│ lucy ┆ ["bookC", "bookC", "bookC"] │
├╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ bob  ┆ ["bookA", "bookA", "bookA"] │
├╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
│ tim  ┆ ["bookA", "bookA", "bookA"] │
└──────┴─────────────────────────────┘
(base) ➜  polars_poetry time poetry run python3 book_agg.py
Traceback (most recent call last):
  File "/Users/anko/code/polars_poetry/book_agg.py", line 18, in <module>
    df = (df.groupby(["user", "book"])
  File "/Users/anko/Library/Caches/pypoetry/virtualenvs/polars-poetry-7pvLRBPw-py3.9/lib/python3.9/site-packages/polars/internals/frame.py", line 4611, in agg
    wrap_df(self._df)
  File "/Users/anko/Library/Caches/pypoetry/virtualenvs/polars-poetry-7pvLRBPw-py3.9/lib/python3.9/site-packages/polars/internals/lazy_frame.py", line 372, in collect
    return pli.wrap_df(ldf.collect())
RuntimeError: Any(OutOfBounds("take indices are out of bounds"))
poetry run python3 book_agg.py   user: 0.41s system: 0.81s cpu: 341% total clock time: 0.356s
avg shared (code):         0 KB
avg unshared (data/stack): 0 KB
total (sum):               0 KB
max memory:                43872 KB
page faults from disk:     0
other page faults:         7768
```

basically if i run it without the time command, it outputs a list in fav_book instead of a single entry

weirdly if i time it, it seems to somehow modify the behaviour and throw a stack trace (keeping in case it's relevant!)

#### What is the expected behavior?

as per the stack overflow, it used to print

```
shape: (3, 2)
┌──────┬──────────┐
│ user ┆ fav_book │
│ ---  ┆ ---      │
│ str  ┆ str      │
╞══════╪══════════╡
│ tim  ┆ bookA    │
├╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┤
│ bob  ┆ bookA    │
├╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┤
│ lucy ┆ bookC    │
└──────┴──────────┘
```

notice how the column fav_book is a str not a list [str]

**Comment by ritchie46** [comment]
Thansk! Got a fix.

**Comment by ankopainting** [comment]
total legend, thank you :)
